### PR TITLE
Touch Bug 1243129: Js testing and linting

### PR DESCRIPTION
**Pull request by phlax**
Adds jest and enzyme testing to the test env, and eslint, altho its set to not fail for now

### CODE REVIEW

**Comment by mathjazz** [comment]
I'll pass this to @adngdb and @jotes.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+module.exports = {
+    "extends": [
+	"eslint:recommended",
```
Comment (adngdb) [inline .eslintrc.js]: The indentation in this file is a mess, it mixes 3-spaces, 4-spaces and tabs. :/

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	test: false,
+	browser: false,
+	jest: false,
+	Promise: false,
+	require: false,
+	shortcut: false,
+	sorttable: false,
+        $: false,
+        Pontoon: false,
+        jQuery: false
```
Comment (adngdb) [inline .eslintrc.js]: It's interesting to put this here. I would instinctively put them in each file instead, adding only the globals being used there. Maybe keep just a few exceptions here, like `Promise`. But I find there is value in having, at the top of each file, a list of the globals being used. It makes it easier to spot where something is coming from.

File: .travis.yml [hunk 3]
```diff
@@ -14,9 +14,11 @@ before_script:
 script:
   - pylama pontoon
   - python manage.py collectstatic --noinput
-  - python manage.py test --with-coverage
   - ./node_modules/.bin/webpack
+  - npm test
+  - python manage.py test --with-coverage
   - py.test --cov-append --cov-report=term --cov=. -v
+  - (./node_modules/.bin/eslint .; exit 0)
```
Comment (adngdb) [inline .travis.yml]: I like that, but can you add a comment to notice that it should only be temporary?

File: tests/js/foo.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+test('Post truth', () => {
+    expect(true).toBe(true)
+})
```
Comment (adngdb) [inline tests/js/foo.js]: Damn, we'll have to have the talk about semicolons in JS... As you must have guessed, I'm in favor of semicolons everywhere. The reasons are numerous and many have blogged about excellent reasons why we should use them. Also, this is actually the very first time I'm seeing semicolon-less JS code in my career at Mozilla, and it was surprising. 

So, please, semicolons? :-)

File: .eslintrc.js (hunk 1 above)
Comment (phlax) [reply .eslintrc.js]: ill fix

File: .eslintrc.js (hunk 2 above)
Comment (phlax) [reply .eslintrc.js]: im not sure i agree - esp for such things like new language primitives

File: .travis.yml (hunk 3 above)
Comment (phlax) [reply .travis.yml]: for sure

File: tests/js/foo.js (hunk 4 above)
Comment (phlax) [reply tests/js/foo.js]: k

File: .eslintrc.js (hunk 2 above)
Comment (phlax) [reply .eslintrc.js]: rereading, thats what you said re Promise

**Comment by phlax** [comment]
> I wonder how hard it would be to add a make dockertest-js command or something? Or maybe make dockertest accept some argument to run only Python or JS tests?

this seems like a good time to mention makeyfile 🎉 ! https://github.com/phlax/makeyfile

**Comment by adngdb** [comment]
Oh hey `makeyfile` looks pretty cool! That would indeed advantageously replace our current Makefile. What do you think @mathjazz?

File: tests/js/foo.js (hunk 4 above)
Comment (phlax) [reply tests/js/foo.js]: we need to make sure we are linting for that

Comment (adngdb) [reply tests/js/foo.js]: Isn't that the case with eslint by default?

Comment (phlax) [reply tests/js/foo.js]: not with current settings - i cribbed a pretty relaxed moanset

**Pull request merged**